src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -89,4 +90,18 @@ public void ApplyChanges(MicrobeEditor editor)
         storedOrganelle.SetCustomUpgradeObject(new ChemoreceptorUpgrades(shownChoices[compounds.Selected],
             (float)maximumDistance.Value, (float)minimumAmount.Value, colour.Color));
     }
+
+    public void CompoundChanged(int index)
+    {
+        if (shownChoices?[index] != null && shownChoices?.Any() == true)
```

Comment:
This is not correct. Do you want me to code this right to show how this should be done (I understand it can be a bit cryptic sometimes what jetbrains wants if you aren't using their IDE)?